ddspipe_core/src/cpp/core/DdsPipe.cpp

```diff
@@ -527,7 +531,8 @@ void DdsPipe::create_new_service_nts_(
 }
 
 void DdsPipe::activate_topic_nts_(
-        const utils::Heritable<DistributedTopic>& topic) noexcept
+        const utils::Heritable<DistributedTopic>& topic,
+        const EndpointKind& endpoint_kind /*= EndpointKind::reader*/) noexcept
```

Comment:
Be careful with the following case: an endpoint is discovered while its topic is blocked, then it is added to `current_topics_` only losing the kind information. So if the topic is then allowed, a bridge is wrongly created with the default kind value.

Reply:
Related scenario: subsequent relevant discovered endpoints would not be taken into account, but only the first one (used in DdsBridge constructor).

Reply:
Potential solution: create an EndpointsManager that would keep track of all created endpoints, and also the ones that should be created once enabled.